### ISSUE TITLE
Easyer way to create dynamic DataViews

### ISSUE THREAD

**Issue by vgb1993** [dotnet/machinelearning#5895]
**Is your feature request related to a problem? Please describe.**
In my company we want add ML blocks to our arsenal (made with Blockly) with witch you could train and run models. I've read in the docs and some issues that the model must be defined beforehand declaring a Class with some Attributes. And apparently it's not easy to create a dynamic model. Our idea is to feed SQL DataSets to the Trainer. 

**Describe the solution you'd like**
As a user I would like to define the model based on the shape of the input data. For example, a SQL DataSet, a CSV etc.
After that, each column metadata could be added programatically.

**Describe alternatives you've considered**
Both seem overcomplicated to me:
https://stackoverflow.com/questions/56761728/add-custom-column-to-idataview-in-ml-net
https://stackoverflow.com/questions/66893993/ml-net-create-prediction-engine-using-dynamic-class/66913705#66913705

**Additional context**
Experienced c# developer, new to ML.NET

**Comment by michaelgsharp** [comment]
@luisquintanilla @briacht I have seen this come up several times now, so its obviously something people want. Not sure how it will fit with our priorities, but its definitely something we should look more into.

@vgb1993 could you give an example of what you would like to see/what you are thinking?

**Comment by briacht** [comment]
Yes, I've seen this request quite a few times now too! I think this would be good to investigate

**Comment by jberd126** [comment]
Yes! This problem I encountered today. 

For @vgb1993's points above,

1 - Yes
3 - Yes
5 - Yes, CSV for us

Others are maybe/sure.

**Comment by michaelgsharp** [comment]
Currently the easiest way to do a dynamic dataview is using the Microsoft.Data.Analysis.DataFrame because it can dynamically load in a text file and create the schema automatically and then use that in ML.NET. Take a look at [this](https://www.luisquintanilla.me/posts/mlnet-predictions-no-schema.html) for an example.

That being said, some of the other approaches mentioned above are things we are considering, but don't have a timeline for them as of yet.

**Comment by jberd126** [comment]
@michaelgsharp, I do agree that a `Microsoft.Data.Analysis.DataFrame` may be a solution, the issue is that it cannot be streamed (the entire dataframe has to fit into memory to use it).

Are you aware of any partitioning work on dataframes such as the Python dask library for pandas?

**Comment by michaelgsharp** [comment]
Personally I am not but I haven't really looked into it much. @luisquintanilla are you aware of anything?

Yeah, the memory thing can be an issue. I'm not aware of a workaround for now, but this is something we are keeping in mind for future work.

**Comment by danmcmyler** [comment]
I was hoping this would have been added to 2.0.0.

**Comment by rzechu** [comment]
I am using only Microsoft.AutoML and I am able to do workaroud by load data dynamically with loading from SQL, and using input columns as input/labels,column names. 
I also use "sql placeholders like"
SELECT '0' as InputOrLabelColumn /* for text */
SELECT CAST(0 as real) as InputOrLabelColumn /*for numbers */

And loading IDataView from normal SQL query
SELECT Text FROM Table

Works for training AutoML API training and prediction. 
I never used casual Featurize/Fit/Transform methods yet

**Comment by BurnOutTrader** [comment]
Not sure how many people struggled with this but i managed to solve this problem git@github.com:BurnOutTrader/DynamicMLPipeline.git
